From f086c5659cd54946b618ae4c695a8c05096f267a Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Mon, 20 Dec 2021 15:01:36 +0100
Subject: [PATCH 111/137] PCI: qcom: add IPQ8074 Gen3 support

IPQ8074 has one Gen2 and one Gen3 port, Gen3 port is the same one as
in IPQ6018, so reuse the support but just add the missing clocks.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 drivers/pci/controller/dwc/pcie-qcom.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

--- a/drivers/pci/controller/dwc/pcie-qcom.c
+++ b/drivers/pci/controller/dwc/pcie-qcom.c
@@ -184,7 +184,7 @@ struct qcom_pcie_resources_2_7_0 {
 };
 
 struct qcom_pcie_resources_2_9_0 {
-	struct clk_bulk_data clks[5];
+	struct clk_bulk_data clks[7];
 	struct reset_control *rst;
 };
 
@@ -1294,8 +1294,10 @@ static int qcom_pcie_get_resources_2_9_0
 	res->clks[0].id = "iface";
 	res->clks[1].id = "axi_m";
 	res->clks[2].id = "axi_s";
-	res->clks[3].id = "axi_bridge";
-	res->clks[4].id = "rchng";
+	res->clks[3].id = "ahb";
+	res->clks[4].id = "aux";
+	res->clks[5].id = "axi_bridge";
+	res->clks[6].id = "rchng";
 
 	ret = devm_clk_bulk_get(dev, ARRAY_SIZE(res->clks), res->clks);
 	if (ret < 0)
@@ -1698,6 +1700,7 @@ static const struct of_device_id qcom_pc
 	{ .compatible = "qcom,pcie-sdm845", .data = &ops_2_7_0 },
 	{ .compatible = "qcom,pcie-sm8250", .data = &ops_1_9_0 },
 	{ .compatible = "qcom,pcie-ipq6018", .data = &ops_2_9_0 },
+	{ .compatible = "qcom,pcie-ipq8074-gen3", .data = &ops_2_9_0 },
 	{ }
 };
 
